app/src/main/java/org/mozilla/tv/firefox/webrender/cursor/CursorEventRepo.kt

```diff
@@ -0,0 +1,113 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.tv.firefox.webrender.cursor
+
+import android.view.KeyEvent
+import android.view.KeyEvent.ACTION_DOWN
+import android.view.KeyEvent.KEYCODE_DPAD_DOWN
+import android.view.KeyEvent.KEYCODE_DPAD_LEFT
+import android.view.KeyEvent.KEYCODE_DPAD_RIGHT
+import android.view.KeyEvent.KEYCODE_DPAD_UP
+import io.reactivex.Observable
+import io.reactivex.rxkotlin.Observables
+import io.reactivex.subjects.PublishSubject
+import io.reactivex.subjects.Subject
+import org.mozilla.tv.firefox.ScreenController
+import org.mozilla.tv.firefox.ScreenControllerStateMachine
+import org.mozilla.tv.firefox.utils.Direction
+import java.util.concurrent.TimeUnit
+
+private val DIRECTION_KEYS = listOf(
+        KEYCODE_DPAD_UP,
+        KEYCODE_DPAD_DOWN,
+        KEYCODE_DPAD_LEFT,
+        KEYCODE_DPAD_RIGHT
+)
+
+/**
+ * This class exposes low level cursor movement events.
+ *
+ * This is usually not the class you want to use. These are unprocessed, and for
+ * most use cases you will want to use a more abstract class.
+ */
+class CursorEventRepo(screenController: ScreenController) {
+
+    sealed class CursorEvent {
+        /**
+         * This represents the user moving the cursor to the edge of the screen,
+         * attempting to scroll, and failing because the website has no more
+         * content in that direction.
+         */
+        data class ScrolledToEdge(val edge: Direction) : CursorEvent()
```

Comment:
Nit: this is only for top/bottom edge right? `scrolledToVerticalEdge`?

Reply:
I feel like the better solution here would just be to have the repo emit events on any edge.  Our VMs aren't using them right now, but the repo shouldn't know about that, right?  Right now that feels like an odd bit of random coupling.